src/Compiler.Dynamic/PageCompilationOptions.cs

```diff
@@ -12,19 +13,36 @@ public PageCompilationOptions()
     }
 
     public bool IsDebug { get; set; }
+
     internal IFileProvider WebFormsFileProvider { get; set; } = default!;
 
-    internal Dictionary<string, Func<string, BaseTemplateParser>> Parsers { get; } = new(StringComparer.OrdinalIgnoreCase);
+    internal Dictionary<string, Func<string, ICompiledTypeAccessor, DependencyParser>> Parsers { get; } = new(StringComparer.OrdinalIgnoreCase);
 
-    internal void AddParser<TParser>(string extension)
+    internal void AddParser<DParser, TParser>(string extension)
```

Comment:
Instead of both the DependencyParser and the TemplateParser, can we set it up so that we can get the TemplateParser from the DependencyParser, similar to how I set up the TemplateParser to give us a generator? That way, we start from the dependency parser and it can "know" what type it should create

Reply:
yes changed that way..